### ISSUE TITLE
Treeshaking is defeated by parenthesis. 

### ISSUE THREAD

**Issue by mhevery** [rollup/rollup#1128]
main.js
```
import {VERSION} from './core.js';
```

`core.js`
```
function makeVersion(full) {
    return Object.create(Version.prototype, {
        full: { value: full }
    });
}
var Version = makeVersion;

export var VERSION = new Version('0.0.0');
```

Results in an empty file. which is correct. Now change single line like so:

```
var Version = (makeVersion); // add parenthesis
```

And now the result is:
```
'use strict';

function makeVersion(full) {
    return Object.create(Version.prototype, {
        full: { value: full }
    });
}
var Version = (makeVersion);


var VERSION = new Version('0.0.0');
```

I would expect that adding of parenthesis should not change the result of tree shaking.

**Comment by kzc** [comment]
See #1101 for related analysis.

**Comment by kzc** [comment]
Fix in #1138

**Comment by Rich-Harris** [comment]
Released 0.37 with the fix – thanks

**Comment by mhevery** [comment]
thank you.
